### PR TITLE
fix: handle tagline field on ncps hosted buttons

### PR DESCRIPTION
**Pull request by imbrian**
### Description

Hosted Buttons are not able to disable the `tagline` on horizontal layout buttons. This change uses the `tagline` value from `/ncp/api/form-fields`. Defaulting to `false` if the field is not provided.

### Why are we making these changes?

Bug fix: LI-45315

### Screenshots
Before:
tagline-before-screenshot

With fix:
tagline-screenshot

### CODE REVIEW

**Review by spencersablan — APPROVED** [review]
Nice! 🚀

File: src/hosted-buttons/index.test.js [hunk 1]
```diff
@@ -150,5 +155,66 @@ describe("HostedButtons", () => {
...
+      render: renderMock,
+      isEligible: vi.fn(() => false),
+    }));
+    // $FlowIssue
+    getButtonsComponent.mockImplementationOnce(() => Buttons);
+    const HostedButtons = getHostedButtonsComponent();
+    // $FlowIssue
+    request.mockImplementationOnce(() =>
+      // eslint-disable-next-line compat/compat
+      Promise.resolve({
```
Comment (jshawl) [inline src/hosted-buttons/index.test.js:172]: can this reuse the previous response from https://github.com/paypal/paypal-checkout-components/blob/71026ef07f05bd150e29a9e0463d5ea5553b2768/src/hosted-buttons/index.test.js#L34 ?

File: src/hosted-buttons/index.test.js [hunk 2]
```diff
@@ -150,5 +155,66 @@ describe("HostedButtons", () => {
...
+                value: "false",
+              },
+            ],
+          },
+        },
+      })
+    );
+    await HostedButtons({
+      hostedButtonId: "B1234567890",
+      fundingSources: ["paypal", "venmo"],
```
Comment (jshawl) [inline src/hosted-buttons/index.test.js:207]: I think `fundingSources` can be removed here

File: src/hosted-buttons/index.test.js [hunk 3]
```diff
@@ -150,5 +155,66 @@ describe("HostedButtons", () => {
...
+    }).render("#example");
+    expect(Buttons).toHaveBeenCalledWith(
+      expect.objectContaining({
+        hostedButtonId: "B1234567890",
+        style: expect.objectContaining({ tagline: false }),
+      })
+    );
+    expect(Buttons).toHaveBeenCalledTimes(2);
+    expect(renderMock).toHaveBeenCalledTimes(0);
+    expect.assertions(3);
```
Comment (jshawl) [inline src/hosted-buttons/index.test.js:217]: awesome test coverage!

File: src/hosted-buttons/utils.test.js [hunk 4]
```diff
@@ -89,6 +93,107 @@ test("getHostedButtonDetails", async () => {
...
+              value: "vertical",
+            },
+            {
+              name: "color",
+              value: "gold",
+            },
+            {
+              name: "button_text",
+              value: "paypal",
+            },
```
Comment (jshawl) [inline src/hosted-buttons/utils.test.js:181]: ^ similar comment about reusing the previous response variables

File: src/hosted-buttons/utils.test.js [hunk 5]
```diff
@@ -89,6 +93,107 @@ test("getHostedButtonDetails", async () => {
...
+          ],
+        },
+      },
+    })
+  );
+  await getHostedButtonDetails({
+    hostedButtonId,
+    fundingSources: [],
```
Comment (jshawl) [inline src/hosted-buttons/utils.test.js:189]: I think `fundingSources` can be removed here

**Review by jshawl — APPROVED** [review]
Looks good! A few small non-blocking suggestions

File: src/hosted-buttons/index.test.js (hunk 1 above)
Comment (imbrian) [reply src/hosted-buttons/index.test.js:172]: Yeah, thought about it, but the tests are changing the value for `tagline` (true, false, undefined). Felt a new object was cleaner than nested spreads and mapping over the array to change one value.

File: src/hosted-buttons/utils.test.js (hunk 5 above)
Comment (nikrom17) [reply src/hosted-buttons/utils.test.js:189]: Yeah, I would remove these. This is how we are temporarily rendering the v2 experience but will remove the `fundingSources` in this [ticket](https://paypal.atlassian.net/browse/DTPPCPSDK-2229) that's currently in the sprint.

The tagline only gets rendered in the horizontal smart stack and the v2 experience is rendering 2 standalone buttons.

Comment (imbrian) [reply src/hosted-buttons/utils.test.js:189]: It's currently required on the type. Are we removing it completely in that ticket? Might make it optional for now, and @nbierdeman can address in that ticket?

**Pull request merged**